ops/testing.py

```diff
@@ -1509,27 +1509,75 @@ def _get_backend_calls(self, reset: bool = True) -> List[Tuple[Any, ...]]:
             self._backend._calls.clear()
         return calls
 
-    def add_model_secret(self, owner: AppUnitOrName, content: Dict[str, str]) -> str:
-        """Add a secret owned by the remote application or unit specified.
-
-        This is named :code:`add_model_secret` instead of :code:`add_secret`
-        to avoid confusion with the :meth:`ops.Application.add_secret`
-        and :meth:`ops.Unit.add_secret` methods used by secret owner
-        charms.
+    def add_charm_secret(self, owner: AppUnitOrName, content: Dict[str, str]) -> str:
+        """Add a secret owned by the charm.
 
         Args:
-            owner: The name of the remote application (or specific remote
-                unit) that will own the secret.
+            owner: The name of the remote application or a specific remote unit,
+                that will own the secret.
             content: A key-value mapping containing the payload of the secret,
                 for example :code:`{"password": "foo123"}`.
 
         Return:
-            The ID of the newly-secret added.
+            The ID of the newly-added secret.
         """
         owner_name = _get_app_or_unit_name(owner)
         model.Secret._validate_content(content)
         return self._backend._secret_add(content, owner_name)
 
+    def add_model_secret(self, owner: AppUnitOrName, content: Dict[str, str]) -> str:
+        """An alias for :meth:`add_charm_secret` for backward compatibility."""
+        warnings.warn("add_model_secret is deprecated, use add_charm_secret instead.",
+                      category=DeprecationWarning)
+        return self.add_charm_secret(owner, content)
+
+    def add_user_secret(self, content: Dict[str, str]) -> str:
+        """Add a secret owned by the user, simulating the ``juju add-secret`` command.
+
+        Args:
+            content: A key-value mapping containing the payload of the secret,
+                for example :code:`{"password": "foo123"}`.
+
+        Return:
+            The ID of the newly-added secret.
+
+        Example usage (the parameter ``harness`` in the test function is
+        a pytest fixture that does setup/teardown, see :class:`Harness`)::
+
+            # config.yaml
+            options:
+              mysec:
+                type: secret
+                description: "tell me your secrets"
```

Comment:
We should use the more modern charmcraft approach.

```suggestion
            # charmcraft.yaml
            config:
              options:
                mysec:
                  type: secret
                  description: "tell me your secrets"
```